src/main/scala/netty/ActorChannelConnector.scala

```diff
@@ -5,34 +5,76 @@ import io.netty.buffer.Unpooled
+    clients: ActorRef[Clients.Control],
+    config: com.typesafe.config.Config,
+    settings: util.SettingStore
+)(using
+    scheduler: Scheduler,
+    ec: Executor
+):
+  private val step     = intSetting("netty.flush.step")
+  private val interval = intSetting("netty.flush.interval-millis")
+  private val maxDelay = intSetting("netty.flush.max-delay-millis")
+  private val flushQ   = new java.util.concurrent.ConcurrentLinkedQueue[Channel]()
+  private val monitor  = Monitor.connector.flush
+
+  scheduler.scheduleOnce(1 second, () => flush())
+
+  scheduler.scheduleWithFixedDelay(1 minute, 1 minute): () =>
+    monitor.config.step.update(step.get())
+    monitor.config.interval.update(interval.get())
+    monitor.config.maxDelay.update(maxDelay.get())
 
   def apply(endpoint: Endpoint, channel: Channel): Unit =
     val clientPromise = Promise[Client]()
     channel.attr(key.client).set(clientPromise.future)
-    val channelEmit = emitToChannel(channel)
-    val monitoredEmit: ClientEmit = (msg: ipc.ClientIn) =>
+    val channelEmit: ClientEmit = (msg: ipc.ClientIn) =>
       endpoint.emitCounter.increment()
-      channelEmit(msg)
-    clients ! Clients.Control.Start(endpoint.behavior(monitoredEmit), clientPromise)
+      emitToChannel(channel, withFlush = endpoint.alwaysFlush)
+    clients ! Clients.Control.Start(endpoint.behavior(channelEmit), clientPromise)
     channel.closeFuture.addListener:
       new GenericFutureListener[NettyFuture[Void]]:
         def operationComplete(f: NettyFuture[Void]): Unit =
           channel.attr(key.client).get.foreach { client =>
             clients ! Clients.Control.Stop(client)
           }
 
-  private def emitToChannel(channel: Channel): ClientEmit =
+  private def intSetting(key: String) =
+    settings.makeSetting(key, config.getInt(key))
+
+  private def emitToChannel(channel: Channel, withFlush: Boolean): ClientEmit =
     case ipc.ClientIn.Disconnect(reason) =>
       channel
         .writeAndFlush(CloseWebSocketFrame(WebSocketCloseStatus(4010, reason)))
         .addListener(ChannelFutureListener.CLOSE)
     case ipc.ClientIn.RoundPingFrameNoFlush =>
       channel.write { PingWebSocketFrame(Unpooled.copyLong(System.currentTimeMillis())) }
-    case in =>
+    case in if withFlush || interval.get() <= 0 =>
       channel.writeAndFlush(TextWebSocketFrame(in.write))
+    case in =>
+      channel.write(TextWebSocketFrame(in.write))
+      flushQ.add(channel)
+
+  private def flush(): Unit =
+    val maxDelayFactor  = maxDelay.get().toDouble / interval.get()
+    var channelsToFlush = step.get().atLeast((flushQ.size * maxDelayFactor).toInt)
+
+    while channelsToFlush > 0 do
+      Option(flushQ.poll()) match
+        case Some(channel) =>
+          if channel.isOpen then channel.eventLoop().execute(() => channel.flush())
+          channelsToFlush -= 1
+        case _ =>
+          channelsToFlush = 0
+
+    val nextInterval =
+      if interval.get() > 0 then interval.get().millis
```

Comment:
`interval.get() > 0` implies that it is possible that `interval.get() == 0`.

It might be a problem, as I see above on line 65 `[...] / interval.get()`

Reply:
yes, set interval-millis = 0 in config or db.setting is currently how we turn this thing off. so line 65 is definitely a divide by zero bug